hpOneView/resources/fc_sans/san_managers.py

```diff
@@ -0,0 +1,142 @@
+                the response may differ from the requested count if the sum of start and count exceed the total number
+                of items, or if returning the requested number of items would take too long
+            query:
+                A general query string to narrow the list of resources returned.
+                The default is no query - all resources are returned.
+            sort:
+                The sort order of the returned data set. By default, the sort order is based
+                on create time, with the oldest entry first.
+
+        Returns:
+            list: A list of SAN managers
+
+        """
+        return self._client.get_all(start=start, count=count, query=query, sort=sort)
+
+    def get(self, id_or_uri):
+        """
+        Retrieves a single registered SAN Manager by id or uri
+
+        Args:
+            id_or_uri: Could be either the SAN Manager resource id or uri.
+
+        Returns:
+            dict: The SAN Manager resource.
+        """
+        return self._client.get(id_or_uri=id_or_uri)
+
+    def update(self, resource, id_or_uri):
+        """
+        Updates a registered Device Manager
+
+        Args:
+            id_or_uri: Could be either the Device manager id or uri.
+            resource (dict): Object to update.
+
+        Returns:
+            dict: The device manager resource.
+        """
+        return self._client.update(resource=resource, uri=id_or_uri)
+
+    def add(self, resource, provider_uri_or_id, timeout=-1):
+        """
+        Adds a Device Manager under the specified provider
+
+        Args:
+            resource: dict object to add
+            provider_uri_or_id: id or uri of provider
+            timeout:
+                Timeout in seconds. Wait task completion by default. The timeout does not abort the operation
+                in OneView, just stop waiting for its completion.
+
+        Returns:
+            dict: Added SAN Manager
+        """
+        add_client = ResourceClient(self._connection, self.PROVIDER_URI)
+        uri = add_client.build_uri(provider_uri_or_id) + "/device-managers"
+        return self._client.create(resource=resource, uri=uri, timeout=timeout)
+
+    def get_provider_uri(self, provider_name):
+        """
+        Gets uri for a specific provider
+
+        Args:
+            name: Name of the provider
+
+        Returns:
+            uri
+        """
+        provider_client = ResourceClient(self._connection, self.PROVIDER_URI)
+        return provider_client.get_by_name(provider_name)['uri']
+
+    def get_default_connection_info(self, provider_name):
+        """
+        Gets default connection info for a specific provider
+
+        Args:
+            name: Name of the provider
+
+        Returns:
+            dict: default connection information
+        """
+        provider_client = ResourceClient(self._connection, self.PROVIDER_URI)
+        return provider_client.get_by_name(provider_name)['defaultConnectionInfo']
```

Comment:
I think you should test before get the key. Because if it does no exist it will raise a `KeyError : 'defaultConnectionInfo' `.
(same issue on line 129)
In this case, the real error that the user should know is that the resource does not exist